src/X.PagedList/PagedListExtensions.cs

```diff
@@ -119,31 +119,32 @@ public static IPagedList<T> ToPagedList<T>(this IQueryable<T> superset, int page
         {
             throw new ArgumentOutOfRangeException($"pageSize = {pageSize}. PageSize cannot be less than 1.");
         }
+
+        if (superset == null)
+        {
+            return StaticPagedList<T>.Empty(pageNumber, pageSize);
+        }
+
+        var supersetCount = superset.Count();
```

Comment:
Please can you reverse this?

This is one the underlying cause of the refactor. When you pass in an `IQueryable` representing a huge set, the `Count()` call becomes expensive. Hence the passing of the `totalSetCount`. So when we pass the `totalSetCount` there is absolutely no need to count the superset. It would not even be correct since `superSet` in that use case is actually the `subset`

Reply:
It should have a ternary operator to take `totalSetCount` when it is available else, do the `.Count()`

Reply:
With your current logic, `Count()` will always be called. We should make it conditional